app/services/data_migrations/backfill_degrees_new_data.rb

```diff
@@ -0,0 +1,72 @@
+module DataMigrations
+  class BackfillDegreesNewData
+    TIMESTAMP = 20220307165548
+    MANUAL_RUN = true
+
+    def change
+      ApplicationQualification.degree.find_each do |qualification|
+        degree_type = degree_type_for(qualification)
+        institution = institution_for(qualification)
+        subject = subject_for(qualification)
+        grade = grade_for(qualification)
+
+        if [degree_type, institution, subject, grade].any?
+          qualification.update(
+            degree_type_uuid: degree_type&.id,
+            degree_institution_uuid: institution&.id,
+            degree_subject_uuid: subject&.id,
+            degree_grade_uuid: grade&.id,
+          )
+        end
+      end
+    end
+
+    def dry_run
+      results = ApplicationQualification.degree.find_each.map do |qualification|
+        degree_type = degree_type_for(qualification)
+        institution = institution_for(qualification)
+        subject = subject_for(qualification)
+        grade = grade_for(qualification)
+
+        {
+          degree_type_uuid: degree_type&.id,
+          degree_institution_uuid: institution&.id,
+          degree_subject_uuid: subject&.id,
+          degree_grade_uuid: grade&.id,
+        }
+      end
+
+      # rubocop:disable Rails/Output
+      puts "Number of Degrees: #{ApplicationQualification.degree.count}"
+      puts "Degree types found: #{results.select { |result| result[:degree_type_uuid].present? }.size}"
+      puts "Degree institutions found: #{results.select { |result| result[:degree_institution_uuid].present? }.size}"
+      puts "Degree subject found: #{results.select { |result| result[:degree_subject_uuid].present? }.size}"
+      puts "Degree grades found: #{results.select { |result| result[:degree_grade_uuid].present? }.size}"
+      # rubocop:enable Rails/Output
+    end
+
+    def degree_type_for(qualification)
```

Comment:
For this (and the methods below, would it be a good idea to consider the `qualification_type_hesa_code`, `subject_hesa_code` etc before falling back to the strings themselves?

Reply:
I am not sure why but we have more nil hesa_codes in the DB than the degree string values. I don’t have the production DB with me to compare locally but the dry run can give some views on what’s the diff comparing the new data degree strings with the DB degree strings and seeing if is better than the hesa codes. Happy to pair or discuss this on slack today :).